### PR TITLE
ability to make ignore antialiasing less strict

### PR DESCRIPTION
**Pull request by DudaGod**
first six commits from this PR - https://github.com/gemini-testing/looks-same/pull/37

In this PR look only at this commit -> 16898b7: feat: add ability to make ignore antialiasing less strict

### CODE REVIEW

File: lib/antialiasing-comparator.js [hunk 1]
```diff
@@ -80,7 +95,7 @@ module.exports = class AntialiasingComparator {
         // if either the darkest or the brightest pixel has more than 2 equal siblings in both images
         // (definitely not anti-aliased), this pixel is anti-aliased
         return (!this._isAntialiased(img1, minX, minY, data) && !this._isAntialiased(img2, minX, minY, data)) ||
-               (!this._isAntialiased(img1, maxX, maxY, data) && !this._isAntialiased(img2, maxX, maxY, data));
+            (!this._isAntialiased(img1, maxX, maxY, data) && !this._isAntialiased(img2, maxX, maxY, data));
     }
 
     _brightnessDelta(color1, color2) {
```
Comment (xrsd) [inline lib/antialiasing-comparator.js]: I know it's not your code, but why not
```
_brightnessDelta(color1, color2) {
   const rgb2y = (r, g, b) => r * 0.29889531 + g * 0.58662247 + b * 0.11448223;
   return rgb2y(color1.R - color2.R, color1.G - color2.G, color1.B - color2.B);
}
```
I think 3 multiplications will be faster than 6. I'm not sure V8 can optimize such things.

If you have image `1000x1000`, you make extra `24 000 000` multiplications.

Comment (xrsd) [reply lib/antialiasing-comparator.js]: I missed the fact, that antialiasing comparator is called only for different pixels, so boost will be very small.

File: index.js [hunk 2]
```diff
@@ -11,9 +11,7 @@ const utils = require('./lib/utils');
 const readPair = utils.readPair;
 const getDiffPixelsCoords = utils.getDiffPixelsCoords;
 
-const JND = 2.3; //Just noticable difference
-                //if ciede2000 >= JND then colors
-                //difference is noticable by human eye
+const JND = 2.3; // Just noticable difference if ciede2000 >= JND then colors difference is noticable by human eye
```
Comment (xrsd) [inline index.js]: 2 words `noticable` – must be `noticeable`

File: README.md [hunk 3]
```diff
@@ -53,20 +65,23 @@ looksSame('image1.png', 'image2.png', {ignoreCaret: true}, function(error, equal
 
 Both `strict` and `ignoreCaret` can be set independently of one another.
 
-Some devices can have different proportion between physical and logical screen resolutions also
-known as `pixel ratio`. Default value for this proportion is 1.
-This param also affects the comparison result, so it can be set manually with `pixelRatio` option.
+### Comparing images with ignoring antialiasing
+
+Some images has difference while comparing because of antialiasing. These diffs will be ignored by default. You can use `ignoreAntialiasing` option with `false` value to disable ignoring such diffs. In that way antialiased pixels will be marked as diffs. Read more about [anti-aliasing algorithm](http://www.eejournal.ktu.lt/index.php/elt/article/view/10058/5000).
```
[the remark this answers is not in the document]
Comment (DudaGod) [reply README.md]: Okay, you wrote the same thing, but in other words. And to my mind your version i not simpler.

**Pull request merged**